pkgs/development/lua-modules/lux-lua.nix

```diff
@@ -0,0 +1,89 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  libgit2,
+  gnupg,
+  libgpg-error,
+  gpgme,
+  lua,
+  isLuaJIT,
+  nix,
+}:
+let
+  luaMajorMinor = lib.take 2 (lib.splitVersion lua.version);
+  luaVersionDir = if isLuaJIT then "jit" else lib.concatStringsSep "." luaMajorMinor;
+  luaFeature = if isLuaJIT then "luajit" else "lua${lib.concatStringsSep "" luaMajorMinor}";
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lux-lua";
+
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "nvim-neorocks";
+    repo = "lux";
+    # NOTE: Lux's tags represent the lux-cli version, which may differ from the lux-lua version
+    rev = "v0.3.14";
+    hash = "sha256-gkUj3eeN0GnHM5sN4SKM/nHeBKe9ifrkg8TZRvA7FlM=";
+  };
```

Comment:
Does someone have a better suggestion for this?
We would ideally want `lux-lua` and `lux-cli` to be in sync (so they use the same `lux-lib` crate versions.

Perhaps I should use set lux-cli's source to `luaPackages.lux-lua.src`, but that doesn't feel right.
Or maybe this can be solved by adding an update script (in a follow-up PR).

Reply:
> Perhaps I should use set lux-cli's source to luaPackages.lux-lua.src, but that doesn't feel right.

Why not? `nixpkgs-review` will catch rebuilds for both packages then

Reply:
Hmm, fair point. The version check will catch an out-of date `version`.